Update fluentd-gcp addon to 1.25.2

ixdy opened the pull request:
**What this PR does / why we need it**: creates a new version of the `fluentd-gcp` image based on the `1.25` version, with newer upstream dependencies pulled in. Same basic idea as #39705.

The definition for `1.21.2` comes from #41862. I'm not sure why `release-1.4` uses two different versions of `fluentd-gcp`.

cc @timstclair 

**Release note**:
<!--  Steps to write your release note:
1. Use the release-note-* labels to set the release note state (if you have access) 
2. Enter your extended release note in the below block; leaving it blank means using the PR title as the release note. If no release note is required, just write `NONE`. 
-->
```release-note
Update fluentd-gcp addon to 1.25.2
```

ixdy committed:
Create fluentd-gcp:1.25.2 with updated dependencies and use it

k8s-reviewable commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.kubernetes.io/review_button.png" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.kubernetes.io/reviews/kubernetes/kubernetes/41863)
<!-- Reviewable:end -->

ixdy commented:
cc @piosz @mwielgus @Crassirostris

timstclair commented on cluster/addons/fluentd-gcp/fluentd-gcp-image/Dockerfile:
```diff
@@ -20,7 +20,7 @@
 # in the Google Developer Console.
 
-FROM gcr.io/google_containers/ubuntu-slim:0.4
+FROM gcr.io/google_containers/ubuntu-slim:0.4.1
```
How does `ubuntu-slim:0.4.1` differ from `ubuntu-slim:0.6` ?

ixdy replied:
0.4.1 is based off https://github.com/kubernetes/contrib/blob/cd4501bdeef07db42f4856987b02f953c32885f3/images/ubuntu-slim/Dockerfile.build.
0.6 is based off https://github.com/kubernetes/ingress/blob/d002ca2f5ed08e8cf8756acae7329aa2ba536cc5/images/ubuntu-slim/Dockerfile.build

It looks like the main difference is that 0.6 doesn't delete bash. So we could probably switch to 0.6 fairly safely.
0.7 removes even more stuff, which makes it a bit riskier IMO.

Crassirostris commented:
@ixdy 

> I'm not sure why release-1.4 uses two different versions of fluentd-gcp.

That's an error, you can leave only `1.25.2`

ixdy commented:
OK, I've updated the `fluentd-gcp:1.25.2` image to be based off of `ubuntu-slim:0.6`, and I've also updated the configs to use `1.25.2` in both places.

Crassirostris commented:
Thanks! LGTM

timstclair commented:
/lgtm

jessfraz commented:
merging since all green
